arbitrator/prover/src/print.rs

```diff
@@ -301,3 +309,250 @@ impl Display for Module {
+
+            let Ok(mut item_reader) = elem.items.get_items_reader() else {
+                continue;
+            };
+            let item_count = item_reader.get_count();
+            w!(
+                "({} {} (i32.const {})",
+                "elem".grey(),
+                format!("{table_index}").pink(),
+                format!("{offset}").mint()
+            );
+            for _ in 0..item_count {
+                let Ok(item) = item_reader.read() else {
+                    continue;
+                };
+                let ElementItem::Func(index) = item else {
+                    continue;
+                };
+                write!(f, " {}", self.raw_func_name(index))?;
+            }
+            writeln!(f, ")")?;
+        }
+
+        for limits in self.memories.iter() {
+            let max = match limits.maximum {
+                Some(max) => format!(" {max}"),
+                None => "".to_string(),
+            };
+            wln!(
+                "({} {}{})",
+                "memory".grey(),
+                limits.initial.mint(),
+                max.mint()
+            );
+        }
+
+        for data in &self.datas {
+            let (_, mut init) = match data.kind {
+                DataKind::Active {
+                    memory_index,
+                    offset_expr,
+                } => (memory_index, offset_expr.get_operators_reader()),
+                _ => continue,
+            };
+
+            let offset = match (init.read(), init.read(), init.eof()) {
+                (Ok(Operator::I32Const { value }), Ok(Operator::End), true) => value as usize,
+                _ => continue,
+            };
+
+            let data = String::from_utf8(
+                data.data.iter()
+                    .flat_map(|b| std::ascii::escape_default(*b))
+                    .collect::<Vec<u8>>()
+            ).unwrap();
+            wln!(
+                r#"({} (i32.const {}) "{}")"#,
+                "data".grey(),
+                offset.mint(),
+                data.mint()
+            );
+        }
+
+        wln!("");
+
+        for (i, g) in self.globals.iter().enumerate() {
+            wln!("({} {i} {} ({})", "global".grey(), g.ty().mint(), g.wat_string());
+        }
+
+        for (export_name, (index, kind)) in &self.exports {
+            use ExportKind as E;
+            let (kind, name) = match kind {
+                E::Func => ("func", self.raw_func_name(*index)),
+                E::Table => ("table", format!("{index}")),
+                E::Memory => ("memory", format!("{index}")),
+                E::Global => ("global", format!("{index}")),
+                E::Tag => ("tag", format!("{index}")),
+            };
+            wln!("({} \"{}\" ({} {}))", "export".grey(), export_name.pink(), kind.grey(), name.pink());
+        }
+
+        for (i, type_idx) in self.functions.iter().enumerate() {
+            let i1 = i as u32;
+
+            let export_str = match self.maybe_func_name(i1 + 1) {
```

Comment:
Concerned that incorrect index is being used here

Reply:
Fixed :)